Sources/SwiftSyntax/StringRef.swift

```diff
@@ -0,0 +1,205 @@
+//===---------- StringRef.swift - Unowned String Representation  ----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Darwin)
+import Darwin
+#endif
+
+private func compareMemory(_ s1: UnsafePointer<UInt8>, _ s2: UnsafePointer<UInt8>, _ count: Int) -> Bool {
+  // FIXME: Need performance check.
+  assert(count > 0)
+#if canImport(Darwin)
+  return Darwin.memcmp(s1, s2, count) == 0
+#else
+  return UnsafeBufferPointer(start: s1, count: count)
+    .elementsEqual(UnsafeBufferPointer(start: s2, count: count))
+#endif
+}
+private func copyMemory(_ dst: UnsafeMutablePointer<UInt8>, _ src: UnsafePointer<UInt8>, _ count: Int) {
+  // FIXME: Need performance check.
+  assert(count > 0)
+#if canImport(Darwin)
+  Darwin.memcpy(dst, src, count)
+#else
+  dst.initialize(from: src, count: count)
+#endif
+}
+
+/// Represent a string.
+///
+/// This type does not own the string data. The data reside in some other buffer
+/// whose lifetime extends past that of the StringRef.
+public struct StringRef {
```

Comment:
A nit, but I really dislike the name `StringRef` even in C++. I'd suggest something like `SyntaxString` or `SyntaxText` or `TextSpan`.

Reply:
Let's go with `SyntaxText`